src/common/buffer.cc

```diff
@@ -143,7 +143,19 @@ static ceph::spinlock debug_lock;
     raw_zeros(char *dataptr, unsigned l, int mempool)
       : raw_combined(dataptr, l, mempool) {
       memset(dataptr, 0, l);
-      // TODO: mprotect dataptr
+#ifndef _WIN32
+      if (mprotect(dataptr, l, PROT_READ) != 0) {
+        ceph_abort("mprotect on raw_zeros failed");
+      }
+#endif
+    }
+
+    ~raw_zeros() {
+#ifndef _WIN32
+      if (mprotect(data, len, PROT_WRITE | PROT_READ) != 0) {
```

Comment:
should we abort on a failure of the dtor to turn the W flag back on?
(BTW - there's a spelling error in the message).